addons/purchase/models/purchase_order.py

```diff
@@ -1290,14 +1293,24 @@ def _update_order_line_info(
                 'product_qty': quantity,
                 'sequence': self._get_new_line_sequence(child_field, section_id),
             })
-            if pol.selected_seller_id:
-                # Fix the PO line's price on the seller's one.
-                seller = pol.selected_seller_id
-                price = seller.price_discounted
-                if seller.currency_id != self.currency_id:
-                    price = seller.currency_id._convert(seller.price_discounted, self.currency_id)
-                pol.price_unit = price
-        return pol.price_unit_discounted
+            params = {'order_id': self}
+            seller = pol.product_id._select_seller(
+                partner_id=self.partner_id,
+                quantity=None,
```

Comment:
same `quantity` issue as in `account`, also don't understand why you can't use `selected_seller_id`